src/main/kotlin/racingcar/service/RacingCarGameMachine.kt

```diff
@@ -6,11 +6,17 @@ import racingcar.service.strategy.NumberGenerateStrategy
 
 class RacingCarGameMachine(private val numberGenerate: NumberGenerateStrategy) {
 
-    fun play(carCount: Int, tryCount: Int): List<GameResultDto> {
-        var cars = List(carCount) { Car() }
+    fun play(carNames: List<String>, tryCount: Int): List<GameResultDto> {
+        var cars = carNames.map { name -> Car(name) }
         return (1..tryCount).map { stage ->
             cars = cars.map { car -> car.move(numberGenerate.generate()) }
-            cars.let { cars -> GameResultDto(stage, cars) }
+            GameResultDto(stage, cars)
         }
     }
+
+    fun getWinners(gameResults: List<GameResultDto>): List<String> {
+        val lastStage = gameResults.sortedBy { it.stage }.last()
+        val maxPosition = lastStage.cars.maxOfOrNull { it.position } ?: 0
+        return lastStage.cars.filter { it.position == maxPosition }.map { it.name }
```

Comment:
이미 5단계 요구사항인 `MVC 패턴`으로 잘 분리하신 것 같으니, 다음 단계에서 딱히 개선하실 부분이 없을 것으로 보여요. 😅

추가적으로 `List<Car>`를 랩핑한 일급컬렉션을 만들어보면 어떨까요? 
또는 `시도횟수`와 `자동차들`을 프로퍼티로가지는 `RacingCarGame` 객체를 만들어볼 수도 있을 것 같아요. 😃

```suggestion
        return lastStage.cars.extractByMaxPosition()
```

Reply:
다음 미션떄 반영해볼게요 : ) 감사합니당